txscript/opcode.go

```diff
@@ -2575,7 +2575,16 @@ func opcodeCheckMultiSig(op *opcode, data []byte, vm *Engine) error {
 		return scriptError(ErrTooManyOperations, str)
 	}
 
-	pubKeys := make([][]byte, 0, numPubKeys)
+	// Use an array on the stack for typical sizes to avoid an allocation in the
+	// vast majority of cases.
+	const numPubKeysHint = 5
+	var pubKeys [][]byte
+	if numPubKeys <= numPubKeysHint {
+		var pubKeysArr [numPubKeysHint][]byte
+		pubKeys = pubKeysArr[0:0:numPubKeys]
```

Comment:
I had forgotten this syntax to set the capacity from an existing array. Just confirming: https://go.dev/play/p/rW_AkOcMhSg